runtime/pubsub/internal/gcp/topic.go

```diff
@@ -56,8 +56,9 @@ func (t *topic) Subscribe(logger *zerolog.Logger, _ *types.SubscriptionConfig, s
 
 	// If we have a resource ID, let's register a push endpoint for it
 	if subscriptionCfg.ResourceID != "" {
-		if t.serverCfg.PushServiceAccount != "" {
-			registerPushEndpoint(t.serverCfg, subscriptionCfg, f)
+		serverCfg := t.serverCfg[subscriptionCfg.ServerID].GCP
```

Comment:
Can we put an explicit check to see if `t.serverCfg[subscriptionCfg.ServerID]` exists or not.

Right now  this will panic with a nil pointer if we get this wrong - where as it would be nicer to panic with a error we'd understand

Reply:
I don't think we should validate the internal consistency of our configuration in business logic, but rather in `config.ParseRuntime` where we do some lightweight validation already today.